### DISCUSSION TITLE
Checked checkbox not working when using wire:model

### DISCUSSION THREAD

**Discussion by toffeegz** [livewire/livewire#3634] [1. Help] [answered]
`<input class="form-check-input" type="checkbox" checked wire:model="selectedPermissions">`
but checked is working fine when I remove the wire:model

I am new in livewire so maybe Im just missing something

**Comment by toffeegz** [comment, marked as answer]
I'm sorry. I fixed it now.

**Comment by AB-Omar** [comment]
how did you solved the issue?

**Reply by joshhanley** [reply]
@AB-Omar putting `checked` on the input doesn't work with Livewire, instead you need to pre-populate your property in your class with `true` if you want it to be checked by default.

Hope this helps!

**Reply by AB-Omar** [reply]
@toffeegz  sorry mate I am new to livewire, on the page load it's fine for a while the checkboxes are checked, but suddenly unchecked when the page is loaded successfully.
here is my code 

**<div class="row offset-3">
                    <div class=" col-md-6">
                        @foreach ($abilities as $ability )
                        <div class="clearfix form-group">
                            <div class="icheck-primary d-inline">
                                <input type="checkbox" wire:model.defer="permissions.{{ $ability->id }}" value="{{ $ability->id }}"
                                 @if ($oldPermissions->contains($ability)) ? checked @endif >
                                <label for="todoCheck2"></label>
                                <label for="checkboxPrimary1"> {{ $ability->title }}</label>
                            </div>
                        </div>
                        @endforeach
                    </div>
                </div>**

**Reply by AB-Omar** [reply]
public $role;
    public $title;
    public $name;
    public $permissions = [];
    public $oldPermissions;
    protected $rules = [
        'title' => 'required',
        'name' => 'required',
        'permissions' => '',
    ];

    public function render()
    {
        return view('livewire.admin.roles.edit-role',[
            'abilities' => Permission::get(),
        ])
        ->layoutData([
            'title' => __('Roles'),
            'desc' => __('Edit Role'),
        ]);;
    }

    public function mount(Role $role)
    {
        $this->title = $role->title;
        $this->name = $role->name;

        //To change the array value from integer to string
        $this->permissions = $this->role->permissions()->pluck('id')->map(function ($id) {
            return strval($id);
            })->toArray();

        // dd($this->permissions);
    }

    public function EditRole(){

        $this->validate();
        
        $this->role->update([
            'title' => $this->title,
            'name' => $this->name,
        ]);

        if ($this->permissions) {
          
            $this->role->permissions()->sync($this->permissions);
        }

        return redirect()->route('admin.roles');
    }

**Reply by AB-Omar** [reply]
Sorry, I am new to livewire couldn't able to get your point.

**Reply by joshhanley** [reply]
@AB-Omar with Livewire putting "checked" on the input in the HTML doesn't work.

So on your input here

```html
<input type="checkbox" wire:model.defer="permissions.{{ $ability->id }}" value="{{ $ability->id }}"
@if ($oldPermissions->contains($ability)) ? checked @endif >
```

This code `@if ($oldPermissions->contains($ability)) ? checked @endif` won't work as you are expecting.

The reason is, Livewire populates your checkbox for you when the javascript is loaded on the page.
That is why you are seeing it checked for a moment, but then it then unchecks.

The way you can ensure it is "checked" using Livewire, is to make sure your `permissions` property has the value set to true.

For example if you had a property called `$show` in your component class, you would set it to `true` for it to be checked

```php
$show = true;
```

Then in your html your checkbox input would look like this

```blade
<input type="checkbox" wire:model="show" />
```

When you then load the page, you will see that checkbox is now checked. If you set `$show = false;` then it would not be checked.

Hope this helps!
